tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/PublicAssetSetMethod.scala

```diff
@@ -0,0 +1,133 @@
+                                                  processingContext: ProcessingContext): ProcessingContext =
+    results.retrieveCreated.foldLeft(processingContext)({
+      case (processingContext, (creationId, result)) =>
+        processingContext.recordCreatedId(ClientId(creationId.id), ServerId(Id.validate(result.id.asString()).toOption.get))
+    })
+}
+
+object PublicAssetSetCreatePerformer {
+  val LOGGER: Logger = LoggerFactory.getLogger(classOf[PublicAssetSetCreatePerformer])
+}
+
+class PublicAssetSetCreatePerformer @Inject()(val publicAssetRepository: PublicAssetRepository,
+                                              val blobResolvers: BlobResolvers,
+                                              val publicAssetSetService: PublicAssetSetService) {
+
+  def createPublicAssets(mailboxSession: MailboxSession, request: PublicAssetSetRequest): SMono[PublicAssetCreationResults] =
+    SFlux.fromIterable(request.create.getOrElse(Map.empty))
+      .concatMap {
+        case (publicAssetCreationId: PublicAssetCreationId, publicAssetCreationRequestAsJson: JsObject) => parseCreate(publicAssetCreationRequestAsJson)
+          .fold(SMono.error(_), creationRequest => createPublicAssets(mailboxSession, creationRequest, request.accountId)
+            .map(publicAsset => PublicAssetCreationSuccess(publicAssetCreationId, PublicAssetCreationResponse.from(publicAsset))))
+          .onErrorResume(error => handleError(publicAssetCreationId, error))
+      }.collectSeq()
+      .map(PublicAssetCreationResults)
+
+  private def handleError(publicAssetCreationId: PublicAssetCreationId, error: Throwable): SMono[PublicAssetCreationResult] =
+    error match {
+      case e: PublicAssetException => SMono.just(PublicAssetCreationFailure(publicAssetCreationId, e))
+      case e: BlobNotFoundException => SMono.just(PublicAssetCreationFailure(publicAssetCreationId, e))
+      case e => SMono.error(e)
+    }
+
+  private def createPublicAssets(mailboxSession: MailboxSession,
+                                 creationRequest: PublicAssetSetCreationRequest,
+                                 accountId: AccountId): SMono[PublicAssetStorage] =
+    generatePublicAssetCreationRequest(creationRequest, mailboxSession, accountId)
+      .flatMap(publicAsset => SMono(publicAssetRepository.create(mailboxSession.getUser, publicAsset)))
+
+  private def parseCreate(jsObject: JsObject): Either[PublicAssetCreationParseException, PublicAssetSetCreationRequest] =
+    PublicAssetSetCreationRequest.validateProperties(jsObject)
+      .flatMap(validJsObject => PublicAssetSerializer.deserializePublicAssetSetCreationRequest(validJsObject) match {
+        case JsSuccess(creationRequest, _) => Right(creationRequest)
+        case JsError(errors) => Left(PublicAssetCreationParseException(standardError(errors)))
+      })
+
+  private def parseAndCheckIdentityIds(creationRequest: PublicAssetSetCreationRequest, session: MailboxSession): SMono[Seq[IdentityId]] =
```

Comment:
Why `parseAndCheck`?

Reading the method I have the impression it only 'checks'...